### PR TITLE
Describe how to run a subset of all unit tests

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
Useful for contributors to check their work-in-progress by running only the tests which they expect to be affected by their changes, as the full test suite can take several minutes to execute.

### CODE REVIEW

File: .github/CONTRIBUTING.md
Comment (dnephin) [inline .github/CONTRIBUTING.md:98]:
```suggestion
- `go test -v ./connect` will run all tests in the connect package (see `./connect` folder)
```

File: .github/CONTRIBUTING.md
Comment (dnephin) [inline .github/CONTRIBUTING.md:99]:
```suggestion
- `go test -v -run TestRetryJoin ./command/agent` will run all tests in the `command/agent` package with name substring `TestRetryJoin`
```

**Review by dnephin — APPROVED** [review]
Thanks! LGTM, but one suggestion for improving clarity. Modules in Go are how libraries are distributed. Every module contains one or more packages. The things being referred to here are packages.

File: .github/CONTRIBUTING.md
Comment (dnephin) [reply .github/CONTRIBUTING.md:99]: I also changed the order of the args. I believe the `-run` arg must come before the package list.

Comment (jkirschner-hashicorp) [reply .github/CONTRIBUTING.md]: Turns out it works either way - I had been running the commands this way while working on my first bugfix. But I can see why that wouldn't be expected (flags don't usually come at the end like that), so I 100% agree with this change.

**Comment by jkirschner-hashicorp** [comment]
PR review feedback addressed in a commit that's been fixed-up with the original. Will merge given prior approval once tests pass in CI.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/417101.

**Pull request merged**
